microcosm_flask/tests/conventions/test_upload.py

```diff
@@ -0,0 +1,169 @@
+"""
+Alias convention tests.
+
+"""
+from uuid import uuid4
+
+from hamcrest import (
+    all_of,
+    assert_that,
+    contains,
+    equal_to,
+    has_entry,
+    has_item,
+    has_key,
+    is_,
+    is_not,
+)
+from json import loads
+from marshmallow import fields, Schema
+from microcosm.api import create_object_graph
+from six import b, BytesIO
+
+from microcosm_flask.namespaces import Namespace
+from microcosm_flask.conventions.base import EndpointDefinition
+from microcosm_flask.conventions.swagger import configure_swagger
+from microcosm_flask.conventions.upload import configure_upload
+from microcosm_flask.operations import Operation
+from microcosm_flask.swagger.definitions import build_path
+from microcosm_flask.tests.conventions.fixtures import Person
+
+
+def upload_file(files, extra):
+    pass
+
+
+def upload_file_for(files, person_id):
+    return dict(
+        id=person_id,
+    )
+
+
+class FileExtraSchema(Schema):
+    extra = fields.String(missing="something")
+
+
+class FileResponseSchema(Schema):
+    id = fields.UUID(required=True)
+
+
+UPLOAD_MAPPINGS = {
```

Comment:
I have examples of:
 - `Update` and `UpdateFor`
 - 204 and 200 responses
 - query string parameters